src/eval/array.rs

```diff
@@ -199,6 +200,36 @@ impl Array {
         Ok(acc)
     }
 
+    /// Calculates the sum of the array's items
+    pub fn sum(&self, default: Option<Value>, span: Span) -> SourceResult<Value> {
+        let mut acc = self
+            .first()
+            .map(|x| x.clone())
+            .or(default.ok_or_else(|| {
+                eco_format!("cannot calculate sum of empty array with no default")
+            }))
+            .at(span)?;
```

Comment:
I think this `eco_format!` here will be run every time a default isn't specified. It would probably be best to do something like this instead:
```suggestion
            .or(default)
            .ok_or_else(|| {
                eco_format!("cannot calculate sum of empty array with no default")
            })
            .at(span)?;
```